extension/scripts/background.js

```diff
@@ -12,6 +12,45 @@ if (localStorage.getItem('enabled') === null) {
 	enabled = localStorage.getItem('enabled') === 'true';
 }
 
+let blackListPromise;
+(function updateBlacklist () {
+	blackListPromise = fetch(
+		'http://bertha.ig.ft.com/view/publish/gss/1_f-cR8VbcujCsjxTLyKVRcG9yp_qUbT0-FrdmwBp0OM/blacklist'
+	).then(response => response.json());
+
+	// Once an hour check for new black listed sites.
+	setTimeout(updateBlacklist , 3600 * 1000);
+}());
+
+function checkIsNotBlackListed (url) {
```

Comment:
It's throwing as a way to reject the promise rather than actually having an error. The flow through the code is much neater with Promise rejections than with a true or false.
It makes much more sense to me that it's going through it checks if the site is black listed if it is it rejects with a reason. Which makes the generator reject with that reason. It's a pattern I'd seen in membership and I really like it.